drivers/power/adm1177/adm1177.c

```diff
@@ -0,0 +1,311 @@
+ *          command byte to the slave device.
+ *
+ * @param dev - The device structure.
+ *
+ * @param slave_address - The slave address, by default SLV_ADDR_WR address should
+ *                        be used(0xB4).
+ *
+ * @param command_byte - The command byte indicates which operation to be
+ *                       performed. MSB = 0, and the 7 other LSBs are :
+ *                       C0 - V_CONT - set to convert voltage continuously.
+ *                       C1 - V_ONCE - set to convert voltage once.
+ *                       C2 - I_CONT - set to convert current continuously.
+ *                       C3 - I_ONCE - set to convert current once.
+ *                       C4 - VRANGE - 0 selects a 14:1 divider, V_FULLSCALE
+ *                                     is then set to 26.35V, 1 selects a
+ *                                     7:2 divider, V_FULLSCALE is then set
+ *                                     to 6.65V.
+ *                       C5 - Unused
+ *                       C6 - STATUS_RD - if this bit is set, then the data
+ *                                        byte read back from the ADM1177
+ *                                        is the status byte.
+ *
+ * @return 0 in case of success, -1 otherwise.
+*******************************************************************************/
+int32_t adm1177_write_command_byte(struct adm1177_dev *device,
+				   uint8_t command_byte)
+{
+	return no_os_i2c_write(device->i2c_desc, &command_byte, 1, 1);
+}
+
+/***************************************************************************//**
+ * @brief - The quick command operation allows the master to check if the slave
+ *          is present on the bus.
+ *
+ * @param dev - The device structure.
+ *
+ * @param slave_address - The slave address, by default is 0xA5.
+ *
+ * @param reg_address - MSB set to 1, indicates an extended register write. The
+ *                      two LSBs indicate which of the three extended registers
+ *                      is to be written to. All other bits should be set to 0.
+ *
+ * @param ext_cmd_byte - The extended command byte indicates which operation of
+ *                       the specified register should be used. Each register
+ *                       has its own operations.
+ *
+ * @return 0 in case of success, -1 otherwise.
+*******************************************************************************/
+int32_t adm1177_write_extended_command_byte(struct adm1177_dev *device,
+		uint8_t reg_address,
+		uint8_t ext_cmd_byte)
+{
+	uint8_t data[2] = {reg_address, ext_cmd_byte};
+
+	return no_os_i2c_write(device->i2c_desc, data, 2, 1);
+}
+
+/***************************************************************************//**
+ * @brief - The read status function reads the status byte which has its own
+ *          operations. Status Byte is read only when C6 in command byte is 1.
+ *
+ * @param dev - The device structure.
+ *
+ * @param slave_address - The slave address, by default is 0xA5.
```

Comment:
you don't have this function parameter actually.

Reply:
Fixing it.